src/Castle.Core/DynamicProxy/Contributors/InterfaceProxyWithoutTargetContributor.cs

```diff
@@ -62,6 +63,15 @@ protected override IEnumerable<MembersCollector> CollectElementsToProxyInternal(
 
 		private Type GetInvocationType(MetaMethod method, ClassEmitter emitter)
 		{
+			var methodInfo = method.Method;
+
+			if (canChangeTarget == false && methodInfo.IsAbstract && methodInfo.IsGenericMethod == false && methodInfo.IsGenericMethodDefinition == false)
+			{
+				// We do not need to generate a custom invocation type because no custom implementation
+				// for `InvokeMethodOnTarget` will be needed (proceeding to target isn't possible here):
+				return typeof(InterfaceMethodWithoutTargetInvocation);
+			}
```

Comment:
Generics only matter because there are some asserts somewhere else that trigger for generic methods; I'll need to look at those more closely before allowing generic methods here.

`IsAbstract` may seem unnecessary now, but it becomes important once we support default interface implementations.

Reply:
Do generic methods not already work? Looks like `SetGenericMethodArguments` would be called on this invocation instance if you removed the check.

Reply:
I think they'd work just like that. It's just that existing code (`Assert`) assumes that either a `CompositionInvocation`- or `InheritanceInvocation`-derived type will come along the way, and AFAIK those carry the proxied method's type parameters at the type level. I wouldn't want to just remove those (probably superfluous) asserts without checking if perhaps there _is_ some case I haven't considered for the new invocation type.

Reply:
I added the above change in a slightly modified form, so generic methods are not covered, too.